1. Line 12 prints `i` which stores the value 3 because it is defined using the keyword `var` which means it can be accessed anywhere in the scope of the function it is defined in which in this case is `discountedPrices`
2. Line 12 prints `discountedPrice` which stores the value 150 because it is defined using the keyword `var` which means it can be accessed anywhere in the scope of the function it is defined in which in this case is `discountedPrices`
3. Line 12 prints `finalPrice` which stores the value 150 because it is defined using the keyword `var` which means it can be accessed anywhere in the scope of the function it is defined in which in this case is `discountedPrices`
4. This function returns the array of discounted prices with values [50,100,150]. The function takes the parameter array  `price` ([100,200,300]) and applies a discount of 50% off and returns those discounted values in the array `discounted`. It successfully does so because the variable discounted is defined with keyword `var` which means it can be accessed anywhere in the scope of the function it is defined in.
5. Line 12 causes an error because it tries to access the variable `i` outside of the previous for loop scope it was originally defined in with keyword `let` at line 6
6. Line 13 causes an error because it tries to access the variable `discountedPrice` outside of the previous for loop scope it was originally defined in with keyword `let` at line 7
7. Line 14 prints `finalPrice` which stores the value 150 because it is defined using the keyword `let` in the scope of the function which means it can be accessed anywhere within the same block it was defined in which in this case is the function `discountedPrices`
8. This function successfully returns the array `discounted` with values [50,100,150]. The function takes the parameter array `price` ([100,200,300]) and applies a discount of 50% off and stores those values in an array `discounted`. Since the array `discounted` is defined with keyword `let` in the scope of the entire function, the function successfully returns `discounted`.
9. Line 11 causes an error because it tries to access the variable i outside of the previous for loop scope it was originally defined in with keyword `let` at line 6
10. Line 12 prints `length` which stores the value 3 because it is defined using the keyword `const` which means it can be accessed anywhere in the block where it was declared which in this case is the function discountPrices. Additionally, the function never attempts to reassign `length`
11. This function returns the `discounted` array with values [ 50, 100, 150 ]. Even though discounted was defined with the keyword `const` in the block scope of the function `discountPrices`, it can still be modified by pushing new values as long as it doesn't get reassigned. Therefore, since the return statement is accessing discounted in the same function block it was defined in it successfully returns `discounted`.
12. 
